### ISSUE TITLE
Paypal Browser stuck in the last step

### ISSUE THREAD

**Issue by caijunjieNick** [braintree/braintree_android#745]
### Braintree SDK Version

4.31.0

### Environment

Production

### Braintree dependencies

implementation 'com.braintreepayments.api:paypal:4.31.0'
implementation 'com.braintreepayments.api:paypal-data-collector:4.31.0'

### Describe the bug

After entering the account credentials, I reach the final payment page. However, when I click the button, it shows "One more step" and stays stuck there. The payment interface doesn't disappear.

### To reproduce

private void startPaypal(String token) {
    braintreeClient = new BraintreeClient(requireContext(), token);
    payPalClient = new PayPalClient(requireActivity(), braintreeClient);
    payPalClient.setListener(this);
    payPalDataCollector = new PayPalDataCollector(braintreeClient);
    collectDeviceData();
    myTokenizePayPalAccountWithCheckoutMethod();
} i add paypallistener to the fragment

### Expected behavior

The payment interface should disappear and  the result should be returned via the payment method listener to the method onPayPalSuccess

### Screenshots

![MicrosoftTeams-image](https://github.com/braintree/braintree_android/assets/33259056/2e86e3f6-20b9-440a-8a1a-7f09c2f2c985)

**Comment by caijunjieNick** [comment]
i replaced the <data android:scheme="${applicationId}.braintree"/> with <data android:scheme="custom-url-scheme-1"/>. The issue has been resolved. But why? I just follow the v4.9.0+_MIGRATION_GUIDE.md. Can someone explain it?

**Comment by sshropshire** [comment]
@caijunjieNick did you set a custom `returnUrlScheme` using the `BraintreeClient` constructor to make it work? Whichever scheme you choose, it should match the scheme set in your `AndroidManifest.xml`.

**Comment by caijunjieNick** [comment]
@sshropshire yes, i set it by using the BraintreeClient constructo and in the AndroidManifest.xml at same time

**Comment by sshropshire** [comment]
@caijunjieNick that'll do it yeah. If you like you can post the snippet of your `AndroidManifest.xml` where the intent is declared. It should work using the default manifest placeholder we mention in the [migration guide](https://github.com/braintree/braintree_android/blob/master/v4.9.0%2B_MIGRATION_GUIDE.md#browser-switch).

**Comment by sshropshire** [comment]
Also, I've seen it break in the past when a merchant app's package name contains underscores.

**Comment by caijunjieNick** [comment]
@sshropshire i just set it with "com.your-company.your-app.braintree" (not ${applicationId}.braintree) and also set it in BraintreeClient constructor. I also did some test. It works fine and didn't break.

**Comment by caijunjieNick** [comment]
@sshropshire hello, when i cancel the final payment page and return back to shoppingcart. And then go into the orderconfirmationFragment(the last site, not paypal payment page), the orderconfirmationFragment was not added into activity. And therefor it break because a method needs Context.

**Comment by caijunjieNick** [comment]
@sshropshire hello, i found the problem.  I write 
` braintreeClient = new BraintreeClient(this, MerchantClientTokenProvider());`
` payPalClient = new PayPalClient(this, braintreeClient);`
`payPalClient.setListener(this);`
`collectDeviceData();`
`myTokenizePayPalAccountWithCheckoutMethod()`
in a method startPaypal() and invoke this method in handleMessage() by using 
`final OrderConfirmationFragment myAct = mAct.get();`
`myAct.startPaypal()`.
Right now i just modified:
`myAct.braintreeClient = new BraintreeClient(myAct.requireContext(), token,"com.akami.apotheke.braintree");
myAct.payPalClient = new PayPalClient(myAct, myAct.braintreeClient);
myAct.payPalClient.setListener(myAct);
myAct.payPalDataCollector = new PayPalDataCollector(myAct.braintreeClient);
myAct.collectDeviceData();
myAct.myTokenizePayPalAccountWithCheckoutMethod();`
And doesn't break anymore
